### PR TITLE
[Issue #1, #2, #9] Implement Command Pattern and few requirements

### LINKED ISSUE #1: [Student A] Menu of application
Check instructions from the doc

### LINKED ISSUE #2: [Student B] Downloading text file from the internet
Check instructions from the doc

### LINKED ISSUE #9: [Student B] Close the application
Check instructions from the doc

### PR DESCRIPTION
**Pull request by MichalRac**
closes #1 - in this case, I allowed myself to switch things up a little to make it more "varied" as @piegus32 suggested before 😄
The main menu now implements a simple command pattern, basically allowing us to not have to care about adding each functionality to the switch case and the menu description, only to add it to the list of commands in the constructor.
What do you think about this solution? Feel free to comment comment the hell out of it! 😃 
closes #2 + some additional failsafe checks, worth taking a look at in case I didn't cover everything
closes #9 + cleans up the file

### CODE REVIEW

**Review by piotrkardasz — APPROVED** [review]
Excellent! 💪

File: TextAnalyzer/Main.cs [hunk 1]
```diff
@@ -9,53 +9,40 @@ namespace TextAnalyzer
 {
     class Main
     {
-        int getResponse;
-
         //Delegate WriteMessege.
         Action<string> writeMessege = Writer.WriteMessege;
+        List<ICommand> optionsList = new List<ICommand>();
```
Comment (jagieldarek) [inline TextAnalyzer/Main.cs]: protected readonly Action<string> writeMessege = Writer.WriteMessege;
        protected readonly List<ICommand> optionsList = new List<ICommand>();

File: TextAnalyzer/Main.cs [hunk 2]
```diff
@@ -9,53 +9,40 @@ namespace TextAnalyzer
...
         {
-            writeMessege("1.Type 1");
-            writeMessege("2.Type 2");
-            writeMessege("3.Type 3");
-            writeMessege("4.Type 4");
+            int i = 1;
+            foreach (var command in optionsList)
+            {
+                writeMessege($"{i++}. {command.Description}");
+            }
```
Comment (jagieldarek) [inline TextAnalyzer/Main.cs]: var i = 1;
            foreach (var command in optionsList) writeMessege($"{i++}. {command.Description}");

File: TextAnalyzer/Main.cs [hunk 3]
```diff
@@ -9,53 +9,40 @@ namespace TextAnalyzer
...
             writeMessege("Took: ");
 
-            //Try-catch exception before switching option.
-            //If getRespone different than int32 , return to method.
+            //Try-catch exception before choosing option.
+            //If the read int is not a valid index in options list, retry
             try
             {
-                getResponse = Convert.ToInt32(Console.ReadLine());
```
Comment (jagieldarek) [inline TextAnalyzer/Main.cs:29]: param getRespone -> podmienić na 'response'

File: TextAnalyzer/TextFileFetcher.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+using System.IO;
+
+namespace TextAnalyzer
+{
+    static class TextFileFetcher
+    {
+        private const string FILE_NAME = "text.txt";
+        private const string LINK = "https://s3.zylowski.net/public/input/3.txt";
+
+        // Fetch the text, download if needed
```
Comment (jagieldarek) [inline TextAnalyzer/TextFileFetcher.cs]: /// <summary>
        /// Get Text File from web client.
        /// </summary>
        /// <returns>Co returnuje?</returns>

File: TextAnalyzer/TextFileFetcher.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                    client.DownloadFile(LINK, FILE_NAME);
+                }
+                catch(Exception)
+                {
+                    Console.Write("Could not fetch the file, try again.");
+                    return null;
+                }
+            }
+
+            if (File.Exists(FILE_NAME)) // Failsafe condition
```
Comment (jagieldarek) [inline TextAnalyzer/TextFileFetcher.cs]: Ten komentarz jest zbędny.

File: TextAnalyzer/TextFileFetcher.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            {
+                return File.ReadAllText(FILE_NAME);
+            }
+            return null;
+        }
+
+        public static void CleanupFileIfPossible()
```
Comment (jagieldarek) [inline TextAnalyzer/TextFileFetcher.cs:43]: /// <summary>
        /// Clean up file if it's possible.
        /// </summary>

Dodaj summary.

File: TextAnalyzer/ExitCommand.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    class ExitCommand : ICommand
+    {
+        public string Description
+        {
+            get
+            {
+                return "Exit Program.";
+            }
+        }
+
```
Comment (jagieldarek) [inline TextAnalyzer/ExitCommand.cs]: /// <summary>
        /// Description
        /// </summary>

File: TextAnalyzer/ExitCommand.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Threading;
+using System.IO;
+
+namespace TextAnalyzer
+{
+    class ExitCommand : ICommand
+    {
+        public string Description
```
Comment (jagieldarek) [inline TextAnalyzer/ExitCommand.cs]: public string Description => "Exit Program.";

File: TextAnalyzer/TemplateCommandOption.cs [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TextAnalyzer
+{
+    class TemplateCommandOption : ICommand
+    {
+        public string Description
```
Comment (jagieldarek) [inline TextAnalyzer/TemplateCommandOption.cs]: public string Description => "Template option, writes the text file in the console";

**Review by jagieldarek — CHANGES_REQUESTED** [review]
Code cleanup + summary.

File: TextAnalyzer/Main.cs (hunk 3 above)
Comment (MichalRac) [reply TextAnalyzer/Main.cs:29]: Adjusted 👍

File: TextAnalyzer/TextFileFetcher.cs (hunk 6 above)
Comment (MichalRac) [reply TextAnalyzer/TextFileFetcher.cs:43]: Uważam że dodawanie komentarza który właściwie tylko powtarza nazwę metody jest zbędne.

File: TextAnalyzer/Main.cs (hunk 1 above)
Comment (MichalRac) [reply TextAnalyzer/Main.cs]: Adjusted 👍

File: TextAnalyzer/Main.cs (hunk 2 above)
Comment (MichalRac) [reply TextAnalyzer/Main.cs]: Adjusted, chociaż osobiście uważam że to jest jedynie kwestia konwencji i tego czy w danym projekcie ważniejsza jest czytelność czy zwięzłość.

File: TextAnalyzer/TextFileFetcher.cs (hunk 4 above)
Comment (MichalRac) [reply TextAnalyzer/TextFileFetcher.cs]: Adjusted 👍

File: TextAnalyzer/TextFileFetcher.cs (hunk 5 above)
Comment (MichalRac) [reply TextAnalyzer/TextFileFetcher.cs]: Adjusted 👍

File: TextAnalyzer/ExitCommand.cs (hunk 8 above)
Comment (MichalRac) [reply TextAnalyzer/ExitCommand.cs]: Adjusted 👍

File: TextAnalyzer/TemplateCommandOption.cs (hunk 9 above)
Comment (MichalRac) [reply TextAnalyzer/TemplateCommandOption.cs]: Adjusted 👍

File: TextAnalyzer/ExitCommand.cs (hunk 7 above)
Comment (MichalRac) [reply TextAnalyzer/ExitCommand.cs]: Adjusted 👍

**Pull request merged**